conductor/__beta__/job/job.py

```diff
@@ -0,0 +1,144 @@
+import logging
+import os
+
+import conductor.lib.conductor_submit as conductor_submit
+import conductor.lib.package_utils
+import conductor.lib.api_client
+
+LOG = logging.getLogger(__name__)
+
+class JobError(Exception):
+    pass
+
+class Job(object):
+    
+    def __init__(self):
+    
+        self.upload_paths = []
+        self.software_packages_ids = []
+        self.owner = 'bob'
+        self.priority = 5
+        self.location = ""
+        self.instance_type = "n1-standard-8"
+        self.metadata = {}
+        self.local_upload = True
+        self.auto_retry_policy = {}
+        self.preemptible = True
+        self.chunk_size = 1
+        self.project = "default"
+        self.output_path = ""
+        self.job_title = ""
+        self.docker_image = ""
+        self._dependencies = None
+        self.environment = {}
+        
+        self._dependency_scan_enabled = True
+        self.conductor_job_id = None
+                
+    def _get_task_data(self):
+        pass
+    
+    def _get_frame_range(self):
+        pass
+    
+    def _get_environment(self):
+        
+        packages = [package for package in conductor.lib.api_client.request_software_packages() if package["package_id"] in self.software_packages_ids]
+        return conductor.lib.package_utils.merge_package_environments(packages, base_env=self.environment)
+    
+    def get_output_path(self):
+        return self.output_path
+    
+    def scan_for_dependencies(self):
+        return []

```

Comment:
Did you really mean to always return an empty list? Can you leave a comment explaining why?

Reply:
> Did you really mean to always return an empty list? Can you leave a comment explaining why?

I would assume that this method is here to be overridden.  But instead of a raising a `NotImplementedError` (which would obliviously halt the execution), it  simply returns an empty list (i.e. there are no dependencies). That being said, some docs describing the purpose of this method would certainly be helpful (before it leaves beta).